zds/antispam/spam_detector.py

```diff
@@ -0,0 +1,81 @@
+import logging
+from datetime import datetime
+
+from django.contrib.auth.models import User
+from django.utils.translation import gettext_lazy as _
+
+from zds.antispam.spam_fields import spam_fields
+from zds.antispam.spam_model_manager import SpamModelManager
+from zds.utils.models import Alert
+
+
+class SpamDetector:
+    def __init__(self):
+        self.logger = logging.getLogger(__name__)
+        self.model_manager = SpamModelManager()
+
+    def check_text(self, text, content_type):
+        """
+        Check if a given text is spam for the specified content type.
+        """
+        if not text:
+            return False
+
+        try:
+            prediction = self.model_manager.predict(content_type, [text])[0]
+            if prediction == 0:  # 0 indicates spam
+                self.logger.info(f"✘ Text looks like spam: {text[:30]}...")
+                return True
+            else:
+                self.logger.info(f"✔️ Text doesn't look like spam.")
+                return False
+        except Exception as e:
+            self.logger.error(f"Error during spam detection: {e}")
+            return False
+
+    def send_alert(self, instance, field_name):
+        """
+        Create an alert for a spam-suspect field with detailed context.
+        """
+        try:
+            # Find the spam field configuration for the instance
+            field_config = next(
+                (
+                    config
+                    for config in spam_fields
+                    if isinstance(instance, config["model"]) and config["field"] == field_name
+                ),
+                None,
+            )
+            if not field_config:
+                self.logger.error(f"No spam field configuration found for {type(instance).__name__}.{field_name}")
+                return
+
+            # Extract scope and instance info
+            scope = field_config["scope"]
+            instance_info = field_config["get_instance_info"](instance)
+
+            # Map scope to the correct Alert model field
+            alert_kwargs = {
+                "author": User.objects.get(username="antispam"),
+                "scope": scope,
+                "text": _(f"Potential spam detected in {instance_info}, field '{field_name}'."),
+                "pubdate": datetime.now(),
+            }
```

Comment:
```suggestion
            scope_to_alert_kwargs = {
                "PROFILE": "profile",
                "FORUM": "comment",
                "CONTENT": "content",
            }
            alert_kwargs = {
                "author": User.objects.get(username="antispam"),
                "scope": scope,
                "text": _(f"Potential spam detected in {instance_info}, field '{field_name}'."),
                "pubdate": datetime.now(),
                scope_to_alert_kwargs[scope]: instance,
            }
```
Mais peut-être qu'on peut définir le scope directement à la valeur attendue par `Alert` ? La question c'est : est-ce qu' on a envie/besoin de distinguer les messages sur le forum et les commentaires des contenus pour détecter du spam ?

Reply:
J’ai fait les modifications comme j’ai compris le commentaire, mais je ne suis pas 100 % sûr que c’est bien ce que le commentaire voulait dire.

Reply:
J'ai fait des changements comme j'ai compris le commentaire, je ne suis pas 100% sûre que c'était ce que était signifié par le commentaire